services/lambda/src/it/java/software/amazon/awssdk/services/lambda/IntegrationTestBase.java

```diff
@@ -91,24 +91,45 @@ public static void tearDown() {
     }
 
     private static File setupFunctionZip(String jsFile) throws IOException {
-        InputStream in = IntegrationTestBase.class.getResourceAsStream(jsFile);
+        try {
 
-        File zipFile = File.createTempFile("lambda-cloud-function", ".zip");
-        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(zipFile));
+            InputStream in = IntegrationTestBase.class.getResourceAsStream(jsFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at line 106. The resource is closed at line 111. There are other execution paths that do not contain closure statements, for example, when File.createTempFile throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).